src/vivarium_nih_us_cvd/components/healthcare_visits.py

```diff
@@ -0,0 +1,266 @@
+
+import pandas as pd
+from vivarium.framework.engine import Builder
+from vivarium.framework.event import Event
+from vivarium.framework.population import SimulantData
+
+from vivarium_nih_us_cvd.constants import data_keys, data_values, models
+
+FOLLOWUP_MIN = 3 * 30  # 3 months
+FOLLOWUP_MAX = 6 * 30  # 6 months
+
+
+class HealthcareVisits:
+    """Manages healthcare utilization"""
+
+    configuration_defaults = {}
+
+    @property
+    def name(self):
+        return self.__class__.__name__
+
+    def __init__(self):
+        pass
+
+    def __repr__(self):
+        return "HealthcareVisits"
+
+    def setup(self, builder: Builder) -> None:
+        self.clock = builder.time.clock()
+        self.step_size = builder.time.step_size()
+        self.randomness = builder.randomness.get_stream(self.name)
+
+        # Load data
+        utilization_data = builder.data.load(data_keys.POPULATION.HEALTHCARE_UTILIZATION)
+        background_utilization_rate = builder.lookup.build_table(
+            utilization_data, parameter_columns=["age", "year"], key_columns=["sex"]
+        )
+        self.background_utilization_rate = builder.value.register_rate_producer(
+            "utilization_rate", background_utilization_rate, requires_columns=["age", "sex"]
+        )
+
+        # Add columns
+        self.visit_type_column = data_values.VISIT_TYPE_COLUMN
+        self.scheduled_visit_date_column = data_values.SCHEDULED_VISIT_DATE_COLUMN
+        self.miss_scheduled_visit_probability_column = data_values.MISS_SCHEDULED_VISIT_PROBABILITY_COLUMN
+        columns_created = [
+            self.visit_type_column,
+            self.scheduled_visit_date_column,
+            self.miss_scheduled_visit_probability_column,
+        ]
+        columns_required = [
+            models.ISCHEMIC_STROKE_MODEL_NAME,
+            models.MYOCARDIAL_INFARCTION_MODEL_NAME,
+        ]
+        self.population_view = builder.population.get_view(columns_required + columns_created)
+
+        # Initialize simulants
+        builder.population.initializes_simulants(
+            self.on_initialize_simulants,
+            creates_columns=columns_created,
+            requires_columns=columns_required,
+        )
+
+        # Register listeners
+        builder.event.register_listener("time_step__cleanup", self.on_time_step_cleanup)
+
+    def on_initialize_simulants(self, pop_data: SimulantData) -> None:
+        """All simulants on SBP medication, LDL-C medication, or a history of
+        an acute event (ie intialized in state post-MI or chronic IS) should be
+        initialized with a scheduled followup visit 0-6 months out, uniformly
+        distributed. All simulants initialized in an acute state should be
+        scheduled a followup visit 3-6 months out.
+
+        For simplicity, do not assign background screenings on initialization.
+
+        A burn-in period will allow the sim to start the observed
+        time period with more realistic boundary conditions.
```

Comment:
not super important, but a note: for someone not familiar with the project, not clear what "boundary conditions" refers to.

Reply:
Oh I mean that very generally, as in the "boundary conditions" of the observed simulation.